services/brig/src/Brig/API.hs

```diff
@@ -1488,56 +1484,73 @@ updateRichInfo (uid ::: _ ::: req) = do
         Just ttl -> setStatus status202 (json (DeletionCodeTimeout ttl))
 
-verifyDeleteUser :: JsonRequest VerifyDeleteUser ::: JSON -> Handler Response
-verifyDeleteUser (r ::: _) = do
+verifyDeleteUserH :: JsonRequest VerifyDeleteUser ::: JSON -> Handler Response
+verifyDeleteUserH (r ::: _) = do
     body <- parseJsonBody r
     API.verifyDeleteUser body !>> deleteUserError
     return (setStatus status200 empty)
 
-onboarding :: JSON ::: UserId ::: JsonRequest AddressBook -> Handler Response
-onboarding (_ ::: uid ::: r) = do
+onboardingH :: JSON ::: UserId ::: JsonRequest AddressBook -> Handler Response
+onboardingH (_ ::: uid ::: r) = do
     ab <- parseJsonBody r
     json <$> API.onboarding uid ab !>> connError
 
-getContactList :: JSON ::: UserId -> Handler Response
-getContactList (_ ::: uid) = do
+getContactListH :: JSON ::: UserId -> Handler Response
+getContactListH (_ ::: uid) = do
     contacts <- lift $ API.lookupContactList uid
     return $ json $ (UserIds contacts)
 
--- Deprecated
-
-deprecatedCompletePasswordReset :: JSON ::: PasswordResetKey ::: JsonRequest PasswordReset -> Handler Response
-deprecatedCompletePasswordReset (_ ::: k ::: req) = do
-    pwr <- parseJsonBody req
-    API.completePasswordReset (PasswordResetIdentityKey k) (pwrCode pwr) (pwrPassword pwr) !>> pwResetError
-    return empty
+-- activation
 
--- Utilities
+-- docs/reference/user/activation.md {#RefActivationSubmit}
+activateKeyH :: JSON ::: JsonRequest Activate -> Handler Response
+activateKeyH (_ ::: req) = respFromActivationResponseWithStatus <$> (activate =<< parseJsonBody req)
 
-activate' :: ActivationKey ::: ActivationCode -> Handler Response
-activate' (k ::: c) = activate $ Activate (ActivateKey k) c False
+activateH :: ActivationKey ::: ActivationCode -> Handler Response
+activateH (k ::: c) = respFromActivationResponseWithStatus <$> (activate (Activate (ActivateKey k) c False))
 
-activate :: Activate -> Handler Response
+activate :: Activate -> Handler ActivationResponseWithStatus
 activate (Activate tgt code dryrun)
     | dryrun = do
         API.preverify tgt code !>> actError
-        return empty
+        return $ ActivationResponseNothing200
     | otherwise = do
         result <- API.activate tgt code Nothing !>> actError
         return $ case result of
             ActivationSuccess ident first -> respond ident first
-            ActivationPass                -> setStatus status204 empty
+            ActivationPass                -> ActivationResponseNothing204
   where
-    respond (Just ident) first = setStatus status200 $ json (ActivationResponse ident first)
-    respond Nothing      _     = setStatus status200 empty
+    respond (Just ident) first = ActivationResponseJust $ ActivationResponse ident first
+    respond Nothing      _     = ActivationResponseNothing200
+
+data ActivationResponseWithStatus
+    = ActivationResponseJust ActivationResponse
+    | ActivationResponseNothing200
```

Comment:
What do you think about not putting status codes here, but more descriptive words instead? But I'm not sure I understand what the different cases exactly mean and why they have those specific status codes.

Reply:
097e188a